modyn/config/schema/pipeline/trigger/cost/cost.py

```diff
@@ -0,0 +1,74 @@
+"""Cost aware triggers are evaluate the trade-off between the cost of a trigger
+(e.g. measured by training machine time) and the benefit of the trigger (e.g.
+measured by a regret metric like data incorporation latency or avoidable
+misclassification incorporation latency)."""
+
+from typing import Literal
+
+from pydantic import Field
+
+from modyn.config.schema.base_model import ModynBaseModel
+
+from ..performance.performance import _InternalPerformanceTriggerConfig
+
+
+class _CostTriggerConfig(ModynBaseModel):
+    """Base class for cost aware trigger policies."""
+
+    cost_tracking_window_size: int = Field(
+        description="How many trigger into the past should be considered for the linear cost w.r.t. data amount model."
+    )
+
+
+class DataIncorporationLatencyCostTriggerConfig(_CostTriggerConfig):
+    """Cost aware trigger policy configuration that uses the data incorporation
+    latency as a regret metric.
+
+    While no trigger occurs the un-integrated data is cumulated and the sum of all seconds of every unincorporated data
+    point is added up.
+
+    `incorporation_delay_per_training_second` servers as a conversion rate between budget (training time) and regret
+    metric (incorporation latency).
+
+    When the regret metric exceeds the training-time based budget metric, a trigger is fired.
+    """
+
+    id: Literal["DataIncorporationLatencyCostTrigger"] = Field("DataIncorporationLatencyCostTrigger")
+
+    # Conversion rate between budget (training time) and regret metric (misclassifications)
+    incorporation_delay_per_training_second: float = Field(
+        description=(
+            "How many seconds of samples having spent in the incorporation queue are we willing to accept "
+            "per second of training time saved."
+        )
+    )
+
+
+class AvoidableMisclassificationCostTriggerConfig(_CostTriggerConfig, _InternalPerformanceTriggerConfig):
+    """Cost aware trigger policy configuration that using the number of
+    avoidable misclassifications integration latency as a regret metric.
```

Comment:
What I don't understand is whether your regret here is just the number of avoidable misclassificaitons (cumulated) or something that also takes into account time (in seconds or sth) because you say latency

Reply:
I suspect the cumulated number of misclassifications is very unstable and badly conditioned on user input. That's why we penalize every second that an avoidable misclassification remains unaddressed (no trigger). It's a result of combining this data incorporation latency idea with the static number of misclassification performance trigger.

Reply:
Okay, maybe you could add exactly this explanation to the docstring. " avoidable misclassifications integration latency" contains this but honestly this comment helps much in understanding it.

Reply:
However, if this is cost-performance, and data incorporation is cost-only, maybe we should not keep them in the same file, but instead have a file for cost-only triggers and one for cost-performance aware triggers?

Reply:
I think for the config models this is fine, cost is the overarching theme for both of this flavours. The respective trigger will both be subclasses of `CostTrigger` but within their own files.